docs/source/audio_dataset_script.mdx

```diff
@@ -0,0 +1,337 @@
+
+    For the `audio` and `text` data at these URLs, use the [`DownloadManager.extract`] method to extract the file paths.
+
+2. After you've downloaded the dataset, use the [`SplitGenerator`] to organize the audio files and text transcriptions in each split. Name each split with a standard name like: `Split.TRAIN`, `Split.TEST`, and `SPLIT.Validation`. 
+
+    In the `gen_kwargs` parameter, specify the file paths to the `audio` and `text` files to load. If necessary, you can use [`DownloadManager.iter_archive`] to iterate over audio files in TAR archives, which allows your dataset to be streamable. The `audio` and `text` file paths are actually passed onto the next step where you'll actually generate the dataset.
+
+<Tip warning={true}>
+
+To stream a TAR archive file, you need to use [`DownloadManager.iter_archive`]! The [`DownloadManager.download_and_extract`] function does not support TAR archives in streaming mode.
+
+</Tip>
+
+```py
+def _split_generators(self, dl_manager):
+    langs = (
+        _ALL_CONFIGS
+        if self.config.name == "all"
+        else [self.config.name]
+    )
+
+    archive_path = dl_manager.download_and_extract(self.config.data_url)
+    audio_path = dl_manager.extract(
+        os.path.join(archive_path, "MInDS-14", "audio.zip")
+    )
+    text_path = dl_manager.extract(
+        os.path.join(archive_path, "MInDS-14", "text.zip")
+    )
+
+    text_path = {l: os.path.join(text_path, f"{l}.csv") for l in langs}
+
+    return [
+        datasets.SplitGenerator(
+            name=datasets.Split.TRAIN,
+            gen_kwargs={
+                "audio_path": audio_path,
+                "text_paths": text_path,
+            },
+        )
+    ]
+```
+
+### Generate the dataset
+
+The last method in the [`GeneratorBasedBuilder`] class actually generates the audio and transcriptions in the dataset. It yields a dataset according to the stucture specified in `features` from the `info` method. As you can see, `generate_examples` accepts the `audio_path` and `text_paths` from the previous method as arguments.
+
+<Tip warning={true}>
+
+To stream a TAR archive file, it needs to be opened and read first. TAR files are accessed and yielded sequentially. This means you need to have the information in the TAR file in hand first so you can yield it with its corresponding audio file.
+
+</Tip>
+
+Now you can write a function for opening and loading examples from the dataset:
+
+```py
+def _generate_examples(self, audio_path, text_paths):
+    key = 0
+    for lang in text_paths.keys():
+        text_path = text_paths[lang]
+        with open(text_path, encoding="utf-8") as csv_file:
+            csv_reader = csv.reader(csv_file, delimiter=",", skipinitialspace=True)
+            next(csv_reader)
+            for row in csv_reader:
+                file_path, transcription, english_transcription, intent_class = row
+
+                file_path = os.path.join(audio_path, *file_path.split("/"))
```

Comment:
Maybe @polinaeterna you know a good example of a dataset with TAR archives instead ? I think this is the main use case for a dataset script for audio right now

Reply:
Good idea! Ideally, the example dataset (maybe something like `vivos`?) should be pretty accessible and straightforward so it's easier for users to follow along 🙂

Reply:
Oh yea `vivos` is totally relevant !